scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/compile/ExpressionCompiler.scala

```diff
@@ -31,31 +31,52 @@ import pl.touk.nussknacker.engine.spel.SpelExpressionParser
 import pl.touk.nussknacker.engine.spel.SpelExpressionParser.Flavour
 import pl.touk.nussknacker.engine.util.Implicits._
 import pl.touk.nussknacker.engine.util.validated.ValidatedSyntax._
+import pl.touk.nussknacker.engine.variables.GlobalVariablesPreparer
 
 object ExpressionCompiler {
 
   def withOptimization(
       loader: ClassLoader,
       dictRegistry: DictRegistry,
       expressionConfig: ExpressionConfigDefinition,
-      classDefinitionSet: ClassDefinitionSet
+      classDefinitionSet: ClassDefinitionSet,
+      expressionEvaluator: ExpressionEvaluator
   ): ExpressionCompiler =
-    default(loader, dictRegistry, expressionConfig, expressionConfig.optimizeCompilation, classDefinitionSet)
+    default(
+      loader,
+      dictRegistry,
+      expressionConfig,
+      expressionConfig.optimizeCompilation,
+      classDefinitionSet,
+      expressionEvaluator
+    )
 
   def withoutOptimization(
       loader: ClassLoader,
       dictRegistry: DictRegistry,
       expressionConfig: ExpressionConfigDefinition,
-      classDefinitionSet: ClassDefinitionSet
+      classDefinitionSet: ClassDefinitionSet,
+      expressionEvaluator: ExpressionEvaluator
   ): ExpressionCompiler =
-    default(loader, dictRegistry, expressionConfig, optimizeCompilation = false, classDefinitionSet)
+    default(
+      loader,
+      dictRegistry,
+      expressionConfig,
+      optimizeCompilation = false,
+      classDefinitionSet,
+      expressionEvaluator
+    )
 
   def withoutOptimization(modelData: ModelData): ExpressionCompiler = {
     withoutOptimization(
       modelData.modelClassLoader.classLoader,
       modelData.designerDictServices.dictRegistry,
       modelData.modelDefinition.expressionConfig,
-      modelData.modelDefinitionWithClasses.classDefinitions
+      modelData.modelDefinitionWithClasses.classDefinitions,
+      ExpressionEvaluator.optimizedEvaluator(
```

Comment:
Based on the comment in `ExpressionEvaluator`, I think that we should use `unOptimizedEvaluator` when we create `ExpressionCompiler` `withoutOptimization` . But ... I'm not sure ;)